docs/proposals/train_api_proposal.md

```diff
@@ -0,0 +1,174 @@
+
+**<h3>Background</h3>**
+
+Currently, there are two flows for data scientists to start using Training operator for their distributed training needs. 
+
+**<h4>Traditional method</h4>**
+
+1. The end-user has to write a custom script that includes data fetching, model loading, and a training loop using pytorch libraries.
+2. Create an image and push it to a registry.
+3. Using the Kubeflow train sdk to create worker spec and master spec which uses the above image and using these two specs create a pytorch job.
+
+This method requires infrastructure knowledge including image creation, registry access and training job spec. This makes it difficult for data scientists to purely focus at the application layer.
+
+**<h4>Higher level SDK</h4>**
+
+To provide a better user experience, a new higher level SDK was added in[ https://github.com/kubeflow/training-operator/pull/1659](https://github.com/kubeflow/training-operator/pull/1659) which avoided the need of image creation. Instead of creating the image, we can directly pass the custom script to train_func parameter and final pytorch job is created
+
+```python
+training_client.create_job(
+   name=pytorchjob_name,
+   train_func=train_function,
+   num_worker_replicas=3, # How many PyTorch Workers will be created.
+)
+```
+
+The above method reduces the time taken but still the end user has to write the train_function method.
+
+In many cases like in LLM, training core code template is a standard and it doesn’t require any custom modification. For faster experimentation of such models, an easier api should exist to train/finetune models without writing custom Python code. HuggingFace Trainer class is a good example but it is not easy to set it up in a Kubernetes environment. Training operator can handle this complexity and provide a better abstracted layer to the user.
+
+**<h3>New Train API proposal:</h3>**
+
+```python
+from kubeflow.training import TrainingClient
+# Arguments related to the model provider including credentials
+model_args = modelProviderClass()
+# Arguments related to the dataset provider including credentials
+dataset_args = datasetProviderClass()
+# Arguments related to the trainer code
+parameters = {key:value pairs}
+trainingClient.train(
+   workers=1, 
+   num_procs_per_worker = 1,
+   resources_per_worker={"gpu": "2", "cpu":8, "memory": "16Gi"}, 
+   model_args, 
+   dataset_args, 
+   parameters
+)
+```
+
+Example: 
+
+```python
+@dataclass
+class HuggingFaceModelParams:
+    access_token = field()
+    model = field()
+
+@dataclass
+class S3DatasetParams:
+    dataset = field()
+    eval_dataset = field()
+    access_token = field()
+    region = field()
+
+@dataclass
+class HuggingFaceTrainParams:
+    learning_rate = field()
+    peft_config = field()
+    transformerClass = field()
+
+trainingClient.train(
+   workers=1, 
```

Comment:
@deepanker13 Let's name this as `num_workers` to be consistent with other parameters and `create_job` API: https://github.com/kubeflow/training-operator/blob/6f2ad5c35aa4042def79f9def1587536585a2302/sdk/python/kubeflow/training/api/training_client.py#L94

```suggestion
   num_workers=1, 
```

Reply:
ok